xcdat/spatial_avg.py

```diff
@@ -433,48 +495,51 @@ def _align_longitude_to_360_axis(self, domain_bounds: xr.DataArray):
 
         Returns
         -------
-        xr.DataArray
-           A DataArray containing the original or upated longitude bounds.
+        Tuple[xr.DataArray, Optional[np.array]]
+           A tuple, with the first element being the domain bounds DataArray,
+           and the second being an np.array with a single element representing
+           the index of the prime meridian grid cell.
 
         Notes
         -----
         This method returns ``domain_bounds`` that are intended for calculating
         spatial weights only.
         """
-
-        # copy bounds to work on
-        d_bounds = domain_bounds.copy()
-        # find instances of d_bounds spanning the prime meridian
-        pmb_index = np.where(d_bounds[:, 1] - d_bounds[:, 0] < 0)[0]
-        # check bounds to ensure they range from 0 to 360
-        if (np.min(d_bounds).values < 0) | (np.max(d_bounds).values > 360):
+        if (domain_bounds.values.min() < 0) | (domain_bounds.values.max() > 360):
             raise ValueError(
-                "Longitude bounds between 0 and 360. Use _swap_lon_axes"
-                "before calling _align_longitude_to_360_axis."
+                "Longitude bounds aren't inclusively between 0 and 360. "
+                "Use `_swap_lon_axes()` before calling `_align_longitude_to_360_axis()`."
             )
 
-        # there should be 0 or 1 such grid cells
-        if len(pmb_index) > 1:
+        p_meridian_index = np.where(domain_bounds[:, 1] - domain_bounds[:, 0] < 0)[0]
+        if len(p_meridian_index) == 0:
+            p_meridian_index = None
+        elif len(p_meridian_index) > 1:
             raise ValueError("More than one grid cell spans prime meridian.")
-        elif len(pmb_index) == 1:
-            # convert index array to integer index
-            pmb_index = pmb_index[0]
```

Comment:
xarray is smart enough to access the equivalent elements without needing to use the integer index, it can just use the np array itself.